ArduPlane/Plane.h

```diff
@@ -787,6 +787,19 @@ class Plane : public AP_Vehicle {
     // terrain disable for non AUTO modes, set with an RC Option switch
     bool non_auto_terrain_disable;
     bool terrain_disabled();
+#if AP_TERRAIN_AVAILABLE
+    bool terrain_enabled_in_current_mode() const;
+    enum class terrain_enabled_bitmask {
+        ALL             = 1U << 0,
+        FLY_BY_WIRE_B   = 1U << 1,
+        CRUISE          = 1U << 2,
+        AUTO            = 1U << 3,
+        RTL             = 1U << 4,
+        AVOID_ADSB      = 1U << 5,
+        GUIDED          = 1U << 6,
+        QRTL            = 1U << 7,
```

Comment:
need LOITER, CIRCLE and maybe THERMAL?
terrain can also be used in QLAND and QLOITER for Plane::relative_ground_altitude

Reply:
I have changed the logic in the param, bit 0 is now a global enable, the remaining bits are per mode disables, this means we no longer have to catch every mode, just add them as required.